docs/value/classwise-shapley.md

```diff
@@ -0,0 +1,145 @@
+and $g$ for an exploration with different base scores. 
+
+??? Example "The default class-wise scorer"
+    ```python
+    import numpy as np
+    from pydvl.value.shapley.classwise import ClasswiseScorer
+    
+    # These are the defaults
+    identity = lambda x: x
+    scorer = ClasswiseScorer(
+        "accuracy",
+        in_class_discount_fn=identity,
+        out_of_class_discount_fn=np.exp
+    )
+    ```
+
+The level curves for $f(x)=x$ and $g(x)=e^x$ are depicted below. The white lines
+illustrate the contour lines, annotated with their respective gradients.
+
+![Level curves of the class-wise utility](img/classwise-shapley-discounted-utility-function.svg)
+
+## Evaluation
+
+We evaluate the method on the nine datasets used in [@schoch_csshapley_2022],
+using the same pre-processing. For images, PCA is used to reduce  down to 32 the
+number of features found by a `Resnet18` model. For more details on the
+pre-processing steps, please refer to the paper.
+
+??? info "Datasets used for evaluation"
+    | Dataset        | Data Type | Classes | Input Dims | OpenML ID |
+    |----------------|-----------|---------|------------|-----------|
+    | Diabetes       | Tabular   | 2       | 8          | 37        |
+    | Click          | Tabular   | 2       | 11         | 1216      |
+    | CPU            | Tabular   | 2       | 21         | 197       |
+    | Covertype      | Tabular   | 7       | 54         | 1596      |
+    | Phoneme        | Tabular   | 2       | 5          | 1489      |
+    | FMNIST         | Image     | 2       | 32         | 40996     |
+    | CIFAR10        | Image     | 2       | 32         | 40927     |
+    | MNIST (binary) | Image     | 2       | 32         | 554       |
+    | MNIST (multi)  | Image     | 10      | 32         | 554       |
+
+
+### Value transfer
+
+We compare CWS to TMCS, Beta Shapley and LOO on a sample removal task with value
+transfer to another model. Values are computed using a logistic regression
+model, and used to prune the training set for a neural network. Random values
+serve as a baseline. The following plot shows valuation-set accuracy of the
+network on the y-axis, and the number of samples removed on the x-axis.
+
+![Accuracy after sample removal using values transferred from logistic regression
+to an MLP](img/classwise-shapley-weighted-accuracy-drop-logistic-regression-to-mlp.svg)
+
+Samples are removed from high to low valuation order and hence we expect a steep
+decrease in the curve. CWS is competitive with the compared methods. Especially
+in very unbalanced datasets, like `Click`, the performance of CWS seems
+superior. In other datasets, like `Covertype` and `Diabetes` and `MNIST (multi)`
+the performance is on par with TMC. For `MNIST (binary)` and `Phoneme` the
+performance is competitive. We remark that for all valuation methods the
+same number of _evaluations of the marginal utility_ was used.
+
+### Value distribution
```

Comment:
If the histograms are to be informative, it can only be when comparing them to the histograms of, say, TMCS or any method. Try adding another method to the plots and discussing the differences.

Reply:
Added comparison to TMCS, as both are competitive.